pkg/generate/generate.go

```diff
@@ -55,30 +50,67 @@ func Generate() error {
 	}).Trace("loaded schema")
 
 	log.WithFields(log.Fields{
-		"packages": cfg.Packages,
+		"count_packages":   len(cfg.Packages),
+		"count_generators": len(cfg.Generators),
 		// "count_mutation":     len(cfg.Mutations),
 		// "count_query":        len(cfg.Queries),
 		// "count_subscription": len(cfg.Subscriptions),
 		// "count_type":         len(cfg.Types),
 		// "package":            cfg.Package,
 	}).Info("starting code generation")
 
-	for _, pkg := range cfg.Packages {
-		for _, pkgGenerator := range pkg.Generators {
+	allGenerators := map[string]generator.Generator{
+		// &terraform.Generator{},
+		"typegen": &typegen.Generator{},
+	}
 
-			for generatorName, generator := range generators {
-				if pkgGenerator.Name == generatorName {
-					err = generator.Generate(s, cfg)
-					if err != nil {
-						return fmt.Errorf("unable to generate for provider %T: %s", generatorName, err)
-					}
-					//
-				}
+	for _, pkgConfig := range cfg.Packages {
+		for _, generatorName := range pkgConfig.Generators {
+			ggg, err := generatorByName(generatorName, allGenerators)
+			if err != nil {
+				log.Error(err)
+				continue
+			}
 
+			genConfig, err := generatorConfigByName(generatorName, cfg.Generators)
+			if err != nil {
+				log.Error(err)
+				continue
 			}
 
+			if ggg != nil && genConfig != nil {
+
+				g := *ggg
+
+				err = g.Generate(s, genConfig, &pkgConfig)
+				if err != nil {
+					return fmt.Errorf("unable to generate for provider %T: %s", generatorName, err)
+				}
+			}
 		}
 	}
 
 	return nil
 }
+
+// generatorConfigByName retrieve the *config.GeneratorConfig from the given set or errros.
+func generatorConfigByName(name string, matchSet []config.GeneratorConfig) (*config.GeneratorConfig, error) {
```

Comment:
Not a blocker, but thoughts on attaching an "action/verb" to the front of these functions names? (same question for `generatorByName`)

```suggestion
func getGeneratorConfigByName(name string, matchSet []config.GeneratorConfig) (*config.GeneratorConfig, error) {
```

Reply:
Sure, that's a good idea.